automated install/basic-install.sh

```diff
@@ -1388,27 +1406,61 @@ check_service_active() {
-      # Note that this breaks dns functionality on host until dnsmasq/ftl are up and running
-      echo -en "${OVER}  ${TICK} Disabling systemd-resolved DNSStubListener"
-      # Make a backup of the original /etc/systemd/resolved.conf
+#  echo -en "  ${INFO} Testing if systemd-resolved is enabled"
+  # Check if Systemd-resolved's DNSStubListener flag is present
+  if [[ "$systemd_resolved_flag" -gt 0 ]]; then
+    if check_service_active "systemd-resolved"; then
+      # Check if DNSStubListener is enabled
+      echo -en "  ${OVER}  ${INFO} Testing if systemd-resolved DNSStub-Listener is active"
+      if ( grep -E '#?DNSStubListener=yes' /etc/systemd/resolved.conf &> /dev/null ); then
+        # Disable the DNSStubListener to unbind it from port 53
+        # Note that this breaks dns functionality on host until dnsmasq/ftl are up and running
+        echo -en "${OVER}  ${TICK} Disabling systemd-resolved DNSStubListener"
+        # Make a backup of the original /etc/systemd/resolved.conf
+        # (This will need to be restored on uninstallation)
+        ${SUDO} sed -r -i.orig 's/#?DNSStubListener=yes/DNSStubListener=no/g' /etc/systemd/resolved.conf
+        echo -e " and restarting systemd-resolved"
+        ${SUDO} systemctl reload-or-restart systemd-resolved
+      else
+        echo -e "${OVER}  ${INFO} Systemd-resolved does not need to be restarted"
+      fi
+    fi
+  else
+    echo -e "${OVER}  ${INFO} Systemd-resolved is not enabled"
+  fi
+}
+
+disable_dnsmasq () {
+  # Check if dnsmasq flag is present.
+  if [[ "$dnsmasq_flag" -gt 0 ]]; then
+    if ( grep -E '#?dns=dnsmasq' /etc/NetworkManager/NetworkManager.conf &> /dev/null ); then
+      # Disable dnsmasq from restarting when Network Manager restarts
+      # Note that this breaks dns functionality on host until FTLDNS is up and running
+      echo -en "${OVER}  ${TICK} Disabling dnsmasq from restarting when Network manager starts/restarts"
+      # Make a backup of the original /etc/NetworkManager/NetworkManager.conf
       # (This will need to be restored on uninstallation)
-      sed -r -i.orig 's/#?DNSStubListener=yes/DNSStubListener=no/g' /etc/systemd/resolved.conf
-      echo -e " and restarting systemd-resolved"
-      systemctl reload-or-restart systemd-resolved
+      ${SUDO} sed -r -i.orig 's/#?dns=dnsmasq/#dns=disabled_dnsmasq/g' /etc/NetworkManager/NetworkManager.conf
+      echo -en "${OVER}  ${TICK} Restarting Network manager"
+      ${SUDO} systemctl reload-or-restart NetworkManager
     else
-      echo -e "${OVER}  ${INFO} Systemd-resolved does not need to be restarted"
+      # Disabling dnsmasq via systemctl
+      echo -e "${OVER}  ${INFO} Disabling dnsmasq via systemct"
+      ${SUDO} systemctl disable dnsmasq &> /dev/null
+      # Running an aditional check (if dnsmasq was loaded via non convetional method)
+      echo -e "${OVER}  ${INFO} Checking if dnsmasq is still running"
+      if [ "$who53" = "dnsmasq" ]; then
```

Comment:
This only checks if dnsmasq was running already, not if it is still running.

Reply:
It actually runs another check for dnsmasq after `systemctl disable dnsmasq &> /dev/null` happens. If dnsmasq was not loaded via `systemct`, the disable command does not work. 
I'm running an additional test for that specific scenario when dnsmasq was loaded via a startup script (user defined or else) and not as a service (basing this on a tested scenario).

Reply:
The `$who53` is not updated as part of the disable line, and it looks like that line will throw a non-zero return code if dnsmasq is not there. If you intend to exit the function if the command returns a non-zero exit code, then explicitly handle that with an if statement.